client/src/pages/ViewRequestStatus/RequestStatusView.js

```diff
@@ -1,16 +1,35 @@
 import React, { useEffect, useState } from "react";
+import io from "socket.io-client";
 // import { Navigate } from "react-router-dom";
 
 import { useParams } from "react-router-dom";
 import ButtonComponent from "./ButtonComponent";
+const socket = io(window.origin, { path: "/api/socket.io" });
 
 export function RequestStatus() {
 	const [request, setRequest] = useState(null);
 	const [donation, setDonation] = useState(null);
 	const [needsReloading, setNeedsReloading] = useState(false);
+	// const [isConnected, setIsConnected] = useState(socket.connected);
 
 	// this helps get the id from the router
 	const { id } = useParams();
+	useEffect(() => {
+		socket.on("connect", () => {
+			console.log("conncted");
+		});
+		socket.on("laptop_request:statusChanged", ({ laptopRequestId }) => {
+			setNeedsReloading((previousNeedsReloading) => {
+				return !previousNeedsReloading;
+			});
+			console.log("statusChanged", laptopRequestId);
+		});
+		socket.emit("test", { requestId: id });
```

Comment:
(-) in the future, `"test"` should become something like `"laptop_request:subscribe"` : a message letting the server know you would like to receive events specifically about this requestId

Reply:
ok, i will change it when i will create the room